src/com/goide/psi/impl/GoPsiImplUtil.java

```diff
@@ -494,6 +494,22 @@ public static GoType resolveType(@NotNull GoTypeReferenceExpression expression)
     return resolve == expression ? new GoCType(expression) : null;
   }
 
+  @Nullable
+  public static GoType resolveUnderlyingType(@Nullable GoCompositeLit literal) {
+    if (literal == null) {
+      return null;
+    }
+
+    GoType type = literal.getType();
+    if (type != null) {
+      return type;
+    }
+
+    GoTypeReferenceExpression ref = literal.getTypeReferenceExpression();
+    GoType resolve = ref != null ? ref.resolveType() : null;
+    return resolve != null ? resolve.getUnderlyingType() : null;
+  }
```

Comment:
It wasn't there when I submitted the pull request. I moved getLiteralType to GoPsiImplUtil.